tools/lib/h5tools.c

```diff
@@ -1025,8 +1034,11 @@ h5tools_fopen(const char *fname, unsigned flags, hid_t fapl_id, bool use_specifi
                 vfd_info.info   = NULL;
                 vfd_info.u.name = drivernames[drivernum];
 
+                /* create a copy of the original fapl */
+                if ((cpy_fapl_id = H5Pcopy(fapl_id)) < 0)
```

Comment:
This seems like a good example of where it becomes awkward to pass a fapl ID in to a function where it's going to close it and you don't necessarily want it to.

Reply:
The other choice is to pass in a DoNotClose bool. This was the only place that really needed it because of the do-loop

Reply:
It seems the simplest just to have the calling code close the ID itself, while just considering `h5tools_get_fapl` as giving out a new ID that's completely separate from the old one. In some cases that looks messier when the old ID needs to be closed, but adding a "DoNotClose" also invites future copy-paste errors.

Reply:
It's messy either way.